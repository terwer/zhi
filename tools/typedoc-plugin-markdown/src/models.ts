import { ReflectionKind, TypeDocOptionMap } from 'typedoc';

/**
 * Defines outputFileStrategy options
 * @enum
 */
export const OutputFileStrategy = {
  Modules: 'modules',
  Members: 'members',
} as const;

export type OutputFileStrategy =
  (typeof OutputFileStrategy)[keyof typeof OutputFileStrategy];

/**
 * Defines format style options
 * @enum
 */
export const FormatStyle = {
  List: 'list',
  Table: 'table',
} as const;

export type FormatStyle = (typeof FormatStyle)[keyof typeof FormatStyle];

/**
 * Defines format style options
 * @enum
 */
export const AnchorFormat = {
  Lowercase: 'lowercase',
  Slug: 'slug',
  None: 'none',
} as const;

export type AnchorFormat = (typeof AnchorFormat)[keyof typeof AnchorFormat];

export interface TypedocPluginMarkdownOptions extends TypeDocOptionMap {
  anchorFormat: AnchorFormat;
  anchorPattern: string;
  baseUrl: string;
  entryDocument: string;
  enumMembersFormat: FormatStyle;
  excludeGroups: boolean;
  flattenOutputFiles: boolean;
  hideBreadcrumbs: boolean;
  hideInPageTOC: boolean;
  hideKindTag: boolean;
  hidePageHeader: boolean;
  hidePageTitle: boolean;
  hideHierarchy: boolean;
  includeFileNumberPrefixes: boolean;
  indentifiersAsCodeBlocks: boolean;
  indexPageTitle: string;
  namedAnchors: boolean;
  outputFileStrategy: OutputFileStrategy;
  propertiesFormat: FormatStyle;
  typeDeclarationFormat: FormatStyle;
}

export interface TemplateMapping {
  directory: string | null;
  template: any;
  kind: ReflectionKind;
}

export type Collapse = 'object' | 'function' | 'all' | 'none';

export type FrontmatterGlobals =
  | string
  | Record<string, string | boolean | number | null>;

export interface NavigationItem {
  title: string;
  url?: string;
  children?: NavigationItem[];
  isReadme?: boolean;
  isGroup?: boolean;
}
