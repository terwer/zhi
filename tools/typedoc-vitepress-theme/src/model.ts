export interface SidebarOptions {
  autoConfiguration: boolean;
  format: 'vitepress' | 'vuepress1' | 'vuepress2';
}
