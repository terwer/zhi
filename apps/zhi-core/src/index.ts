/*
 * Copyright (c) 2023, Terwer . All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Terwer designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Terwer in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Terwer, Shenzhen, Guangdong, China, youweics@163.com
 * or visit www.terwer.space if you need additional information or have any
 * questions.
 */

/**
 * @packageDocumentation
 * zhi-core 主题核心模块
 */

import Zhi from "./theme/zhi"
import DeviceDetection from "zhi-device-detection"

// 主题样式注入入口
import "./style/common/fonts/webfont.css"
import "./style/index.styl"

const loadTheme = async (): Promise<void> => {
  const zhi = new Zhi(DeviceDetection.getDevice())
  await zhi.init()
}

/**
 * 主题入口，由思源笔记自动触发，请勿主动调用
 */
;(async () => {
  await loadTheme()
})()

export default loadTheme
export { Zhi }
