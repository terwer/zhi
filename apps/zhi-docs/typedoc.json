{
  "plugin": [
    "typedoc-plugin-frontmatter",
    "@terwer/typedoc-plugin-markdown",
    "@terwer/typedoc-plugin-markdown"
  ],
  "entryPoints": [
    "../../libs/zhi-device/src/index.ts",
    "../../libs/zhi-env/src/index.ts",
    "../../libs/zhi-log/src/index.ts",
    "../../libs/zhi-common/src/index.ts",
    "../../libs/zhi-blog-api/src/index.ts",
    "../../libs/zhi-fetch-middleware/src/index.ts",
    "../../libs/zhi-xmlrpc-middleware/src/index.ts"
  ],
  "tsconfig": "./tsconfig.json",
  "frontmatterTags": ["author", "description", "outline"],
  "frontmatterGlobals": {
    "outline": "deep"
  }
}