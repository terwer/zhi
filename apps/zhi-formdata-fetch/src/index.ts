import FormdataFetch from "./lib/formdataFetch"
import fetch, { Blob, FormData, Headers } from "node-fetch"

export { FormdataFetch }
export { fetch }
export { Blob, FormData, Headers }
