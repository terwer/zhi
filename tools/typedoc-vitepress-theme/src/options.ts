export const DEFAULT_SIDEBAR_OPTIONS = {
  autoConfiguration: true,
  format: 'vitepress',
};
