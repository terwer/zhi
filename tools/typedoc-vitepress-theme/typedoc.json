{
  "entryPoints": [
    "../../../stubs/src/classes.ts",
    "../../stubs/src/comments.ts"
  ],
  "tsconfig": "../../../stubs/tsconfig.json",
  "plugin": ["typedoc-plugin-markdown", "../dist"],
  "sidebar": { "version": "v1" }
}
