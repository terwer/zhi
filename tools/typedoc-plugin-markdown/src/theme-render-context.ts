import * as path from 'path';
import { Options, Reflection } from 'typedoc';
import { TypedocPluginMarkdownOptions } from './models';
import { partials, templates } from './resources/resources';
import { URL_PREFIX } from './support/constants';
import { MarkdownTheme } from './theme';

export class MarkdownThemeRenderContext {
  public readmeFile = 'README.md';
  public indexFile = 'index.md';
  public modulesFile = 'modules.md';
  public exportsFile = 'exports.md';

  private _activeLocation = '.';

  set activeLocation(activeLocation: string) {
    this._activeLocation = activeLocation;
  }

  get activeLocation() {
    return this._activeLocation;
  }

  constructor(public theme: MarkdownTheme, public options: Options) {}

  getOption<K extends keyof TypedocPluginMarkdownOptions>(name: K) {
    return this.options.getValue(name) as TypedocPluginMarkdownOptions[K];
  }

  getRelativeUrl(url: string | undefined) {
    if (!url) {
      return null;
    }
    if (URL_PREFIX.test(url)) {
      return url;
    } else {
      if (this.getOption('baseUrl')) {
        return this.getOption('baseUrl') + url.replace(/\\/g, '/');
      }

      const relative = path.relative(
        path.dirname(this.activeLocation),
        path.dirname(url),
      );

      return this.parseUrl(
        path.join(relative, path.basename(url)).replace(/\\/g, '/'),
      );
    }
  }

  urlTo(reflection: Reflection) {
    return this.relativeURL(reflection.url);
  }

  parseUrl(url: string) {
    return encodeURI(url);
  }

  relativeURL(url: string | undefined) {
    return this.getRelativeUrl(url);
  }

  templates = templates(this);
  partials = partials(this);
}
