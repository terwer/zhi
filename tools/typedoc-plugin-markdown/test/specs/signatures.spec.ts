import { ProjectReflection, SignatureReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../src/theme-render-context';

describe(`Signatures:`, () => {
  let project: ProjectReflection;
  let context: MarkdownThemeRenderContext;

  beforeAll(async () => {
    ({ project, context } = await global.bootstrap(['signatures.ts'], {
      stubPartials: ['sources'],
    }));
    context.activeLocation = 'signatures.md';
  });

  test(`should compile callable signature'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('CallableSignature') as any).signatures[0],
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with a flag'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('privateFunction') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with params'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithParameters') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function that returns an object'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionReturningAnObject') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a promise that returns an object'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('promiseReturningAnObject') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a promise that returns a symbol'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('promiseReturningASymbol') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function that returns a function'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionReturningAFunction') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with rest params'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithRest') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with optional params'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithOptionalParam') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with union types'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithUnionTypes') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with default values'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithDefaults') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with @return comments'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('commentsInReturn') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile named parameters'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithNamedParams') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile named parameters with comments'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithNamedParamsAndComments') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile pipes in params and comments'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithPipesInParamsAndComments') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function with reference type'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithReferenceType') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function with nested typen params'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('functionWithNestedParams') as any)
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile class with constructor'`, () => {
    expect(
      context.partials.signatureMember(
        (project.getChildByName('ClassWithConstructor') as any).children[0]
          .signatures[0] as SignatureReflection,
        2,
      ),
    ).toMatchSnapshot();
  });
});
