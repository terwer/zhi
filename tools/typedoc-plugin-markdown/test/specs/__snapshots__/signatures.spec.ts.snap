// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signatures: should compile a promise that returns a symbol' 1`] = `
"> **promiseReturningASymbol**(): [\`Promise\`]( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise )\\<\`any\`\\>

[partial: sources]

## Returns

[\`Promise\`]( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise )\\<\`any\`\\>

"
`;

exports[`Signatures: should compile a promise that returns an object' 1`] = `
"> **promiseReturningAnObject**(): [\`Promise\`]( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise )\\<\\{data: \`string\`; id: \`string\`;}\\>

[partial: sources]

## Returns

[\`Promise\`]( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise )\\<\\{data: \`string\`; id: \`string\`;}\\>

"
`;

exports[`Signatures: should compile callable signature' 1`] = `
"> **CallableSignature**(): \`string\`

[partial: sources]

## Returns

\`string\`

"
`;

exports[`Signatures: should compile class with constructor' 1`] = `
"> **new ClassWithConstructor**(x: \`string\`, y: \`string\`): [\`ClassWithConstructor\`](classes/class.ClassWithConstructor.md)

[partial: sources]

## Parameters


| Parameter | Type |
| :------ | :------ |
| x | \`string\` |
| y | \`string\` |


## Returns

[\`ClassWithConstructor\`](classes/class.ClassWithConstructor.md)

"
`;

exports[`Signatures: should compile function that returns a function' 1`] = `
"> **functionReturningAFunction**(): \`Function\`

Comments for function



[partial: sources]

## Returns

\`Function\`

Return comments

> > \\<T\\>(x: \`string\`): \`boolean\`
>
> [partial: sources]
>
> ### Type parameters
>
>
> | Parameter |
> | :------ |
> | T |
>
>
> ### Parameters
>
>
> | Parameter | Type |
> | :------ | :------ |
> | x | \`string\` |
>
>
> ### Returns
>
> \`boolean\`
>
>

"
`;

exports[`Signatures: should compile function that returns an object' 1`] = `
"> **functionReturningAnObject**(): \`object\`

Comments for function



[partial: sources]

## Returns

\`object\`

Return comments

> **x**: \`number\`  = \`1\`

[partial: sources]


> **y**: \`number\`  = \`2\`

[partial: sources]



"
`;

exports[`Signatures: should compile function with nested typen params' 1`] = `
"> **functionWithNestedParams**(params: \`object\`, context: \`any\`): \`boolean\`

Some nested params.

[partial: sources]

## Parameters


| Parameter | Type | Description |
| :------ | :------ | :------ |
| params | \`object\` | The parameters passed to the method. |
| context | \`any\` | The context of the method call. |


## Returns

\`boolean\`

"
`;

exports[`Signatures: should compile function with reference type' 1`] = `
"> **functionWithReferenceType**(descriptor: \`TypedPropertyDescriptor\`\\<\`any\`\\>): \`boolean\`

[partial: sources]

## Parameters


| Parameter | Type |
| :------ | :------ |
| descriptor | \`TypedPropertyDescriptor\`\\<\`any\`\\> |


## Returns

\`boolean\`

"
`;

exports[`Signatures: should compile named parameters with comments' 1`] = `
"> **functionWithNamedParamsAndComments**(\\_\\_namedParameters: \`object\` = \`{}\`, anotherParam: \`string\`): \`void\`

FOO

[partial: sources]

## Parameters


| Parameter | Type | Description |
| :------ | :------ | :------ |
| \\_\\_namedParameters | \`object\` | various options |
| anotherParam | \`string\` | Another param comment |


## Returns

\`void\`

"
`;

exports[`Signatures: should compile named parameters' 1`] = `
"> **functionWithNamedParams**(\\_\\_namedParameters: \`Object\`): \`string\`

[partial: sources]

## Parameters


| Parameter | Type |
| :------ | :------ |
| \\_\\_namedParameters | \`Object\` |


## Returns

\`string\`

"
`;

exports[`Signatures: should compile pipes in params and comments' 1`] = `
"> **functionWithPipesInParamsAndComments**(n: \`number\`): \`number\` \\| \`null\`



[partial: sources]

## Parameters


| Parameter | Type | Description |
| :------ | :------ | :------ |
| n | \`number\` | a\\|b |


## Returns

\`number\` \\| \`null\`

"
`;

exports[`Signatures: should compile signature with @return comments' 1`] = `
"> **commentsInReturn**(): \`boolean\`

Comments with a return definition



[partial: sources]

## Returns

\`boolean\`

Return comments

"
`;

exports[`Signatures: should compile signature with a flag' 1`] = `
"> \`Private\` **privateFunction**(): \`string\`



[partial: sources]

## Returns

\`string\`

"
`;

exports[`Signatures: should compile signature with default values' 1`] = `
"> **functionWithDefaults**(
  valueA: \`string\` = \`'defaultValue'\`, 
  valueB: \`number\` = \`100\`, 
  valueC: \`number\` = \`Number.NaN\`, 
  valueD: \`boolean\` = \`true\`, 
  valueE: \`boolean\` = \`null\`, 
  valueF: \`string\` = \`'<foo>'\`): \`string\`

This is a function with a parameter that has a default value.

[partial: sources]

## Parameters


| Parameter | Type | Default value | Description |
| :------ | :------ | :------ | :------ |
| valueA | \`string\` | 'defaultValue' | A parameter with a default string value. |
| valueB | \`number\` | 100 | A parameter with a default numeric value. |
| valueC | \`number\` | Number.NaN | A parameter with a default NaN value. |
| valueD | \`boolean\` | true | A parameter with a default boolean value. |
| valueE | \`boolean\` | null | A parameter with a default null value. |
| valueF | \`string\` | '\\<foo\\>' | - |


## Returns

\`string\`

"
`;

exports[`Signatures: should compile signature with optional params' 1`] = `
"> **functionWithOptionalParam**(
  firstParamWithDefault: \`boolean\` = \`true\`, 
  requiredParam: \`string\`, 
  optionalParam?: \`string\`, 
  paramWithDefault?: \`number\` = \`0\`): \`void\`

This is a function with a parameter that is optional.

[partial: sources]

## Parameters


| Parameter | Type | Default value | Description |
| :------ | :------ | :------ | :------ |
| firstParamWithDefault | \`boolean\` | true | - |
| requiredParam | \`string\` | undefined | A normal parameter. |
| optionalParam? | \`string\` | undefined | An optional parameter. |
| paramWithDefault? | \`number\` | 0 | - |


## Returns

\`void\`

"
`;

exports[`Signatures: should compile signature with params' 1`] = `
"> **functionWithParameters**(
  paramZ: \`string\`, 
  paramG: \\{prop: \`string\`;} & \\{prop: \`number\`;}, 
  paramA: [\`PropInterface\`](interfaces/interface.PropInterface.md)): \`number\`

This is a function with multiple arguments and a return value.

[partial: sources]

## Parameters


| Parameter | Type | Description |
| :------ | :------ | :------ |
| paramZ | \`string\` | This is a string parameter. |
| paramG | \\{prop: \`string\`;} & \\{prop: \`number\`;} | This is a parameter with intersection type literal. |
| paramA | [\`PropInterface\`](interfaces/interface.PropInterface.md) | This is a **parameter** pointing to an interface. |


## Returns

\`number\`

"
`;

exports[`Signatures: should compile signature with rest params' 1`] = `
"> **functionWithRest**(...rest: \`string\`[]): \`string\`

This is a function with rest parameter.

[partial: sources]

## Parameters


| Parameter | Type | Description |
| :------ | :------ | :------ |
| ...rest | \`string\`[] | The rest parameter. |


## Returns

\`string\`

"
`;

exports[`Signatures: should compile signature with union types' 1`] = `
"> **functionWithUnionTypes**(arg: \`number\` \\| \`boolean\`[], ...args: (\`string\` \\| \`number\`)[]): \`any\`

[partial: sources]

## Parameters


| Parameter | Type |
| :------ | :------ |
| arg | \`number\` \\| \`boolean\`[] |
| ...args | (\`string\` \\| \`number\`)[] |


## Returns

\`any\`

"
`;
